Artforum.js

```diff
@@ -0,0 +1,356 @@
+{
+	"translatorID": "a127f012-4ea4-4d05-a657-24d47f91b016",
+	"label": "Artforum",
+	"creator": "czar",
+	"target": "^https?://(www\\.)?artforum\\.com",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2018-07-08 18:14:25"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2018 czar
+	http://en.wikipedia.org/wiki/User_talk:Czar
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	if (/-\d{5,}$/.test(url)) {
+		return "blogPost";
+	} else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+}
+
+
+function scrape(doc, url) {
+	var translator = Zotero.loadTranslator('web');
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48'); // embedded metadata (EM)
+	translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) { // corrections to EM
+		item.itemType = "blogPost";
+		item.publicationTitle = "Artforum";
+		var json_ld = doc.querySelector('script[type="application/ld+json"]');
+		if (json_ld) {
+			json_ld = json_ld.textContent;
+			item.date = json_ld.match(/"datePublished"\s?:\s?"([^"]*)"/)[1].split(":")[0];
+		}
+		var authorMetadata = doc.querySelectorAll('.contrib-link a');
+		for (let author of authorMetadata) {
+			item.creators.push(ZU.cleanAuthor(author.text, "author"));
+		}
+		if (url.includes('/print/')) {
+			item.itemType = "magazineArticle";
+			item.ISSN = "0004-3532";
+			var issueDate = doc.querySelector('h3.print-article__issue-title');
+			if (issueDate) {
+				item.date = issueDate.textContent.trim().replace('PRINT ','');
+				/* are you feeling adventurous? could follow link to issue's page and scrape the volume/issue numbers...
```

Comment:
If you want to do this, I think the way to go is to reconstruct from the issue description: every year has ten issues from September to May, then a Summer issue, so getting the issue is easy. For the volume, you'd have to do something like 
take the year, subtract 1962, except for September, October, November December, where you subtract 1961. Making the extra call seems tedious, though also doable.

Reply:
i thought about the former but seemed like more of a hack and prone to potential error if the publication cycle was altered in a year

looking for a model: any recommendations for translators that call through to another page?

Reply:
This isn't a perfect example but basically what you'd do is along the lines of https://github.com/zotero/translators/blob/master/Stanford%20Encyclopedia%20of%20Philosophy.js#L71 
You scrape all the info you need from the current page, then do a `ZU.doGet` or (more likely here) `ZU.processDocuments` for the new page add the missing information to the item and (and this is important) finalize the item within that routine, i.e. `item.complete()` would be within the `ZU.processDocuments`.

Reply:
I couldn't perform querySelector/xpath on the doc so had to go with regex—not sure if that's slower